### PR TITLE
Add `lcoLooseMethodDeclaration` option

### PR DESCRIPTION
**Pull request by ineedbots**

### CODE REVIEW

**Comment by ineedbots** [comment]
Allows a directive to enforce parentheses for method declarations. Handy for AUTOINVOKE off mode and for codetools autocomplete to provide the parentheses.

Directive maybe subject to change.

**Comment by ollydev** [comment]
`{$MethodDeclarationParentheses on}` maybe

**Comment by ollydev** [comment]
Add `procedure() of object` `function(): Boolean of object` to your tests too!

File: lpvartypes.pas [hunk 1]
```diff
@@ -36,7 +36,8 @@ interface
     lcoDuplicateLocalNameHints,        //
     lcoArrayHelpers,                   //
     lcoRelativeFileNames,              //
-    lcoInitExternalResult              // Ensure empty result for external calls (useful for ffi)
+    lcoInitExternalResult,             // Ensure empty result for external calls (useful for ffi)
+    lcoMethodDeclarationParentheses    //      {$METHODDECLARATIONPARENTHESES}
```
Comment (ollydev) [inline lpvartypes.pas:40]: Something like ~
// Require method declarations to have `()` if no parameters.

**Pull request merged**